src/firebolt_db/firebolt_async_dialect.py

```diff
@@ -150,8 +152,10 @@ def _init_dbapi_attributes(self) -> None:
             setattr(self, name, getattr(self.dbapi, name))
 
     def connect(self, *arg: Any, **kw: Any) -> AsyncConnectionWrapper:
-
-        connection = await_only(self.dbapi.connect(*arg, **kw))  # type: ignore[attr-defined] # noqa: F821,E501
+        # Synchronously establish a connection that can execute
+        # asynchronous queries later
+        conn_func = partial(self.dbapi.connect, *arg, **kw)  # type: ignore[attr-defined] # noqa: F821,E501
```

Comment:
nit: I believe we can just do `self.dbapi.connect(*arg, **kw)` instead of using partial

Reply:
From my testing `run` accepts only an async function as a first parameter, not a coroutine object that results from `self.dbapi.connect(*arg, **kw)`. It does accept args, but kwargs have to be passed as a partial, I'm afraid.

Reply:
<img width="1118" alt="image" src="https://github.com/firebolt-db/firebolt-sqlalchemy/assets/93913847/ebe16260-ae67-493a-aeee-1edcb8cc1f64">